### DISCUSSION TITLE
change_switch_settings - "use_defaults = configuration" not working as expected

### DISCUSSION THREAD

**Discussion by turi400** [basnijholt/adaptive-lighting#617] [General]
I'm trying to setup an automation to adjust the brightness curve in the evening (I'd rather wait until 7pm for it to start dimming, I'm happy with the color curve).

I'm using adaptive_lighting.change_switch_settings to achieve this. One of my actions is to revert to the settings configured in the AL Integration UI. If my understanding of this service call is correct, using the "use_defaults: configuration" with no other explicit settings should achieve this. Or do I have this completely wrong?

As it stands, if I use the service call below, nothing changes (which is what I would expect if I used the "current" option).

service: adaptive_lighting.change_switch_settings
data:
  entity_id: switch.adaptive_lighting_main
  use_defaults: configuration

The only way I seem to be able to revert without again explicitly setting the values via adaptive_lighting.change_switch_settings is to reload the integration

[home-assistant_adaptive_lighting_2023-06-18T03-09-59.453Z.log](https://github.com/basnijholt/adaptive-lighting/files/11780583/home-assistant_adaptive_lighting_2023-06-18T03-09-59.453Z.log)

[AdaptiveLighting_Configuration.txt](https://github.com/basnijholt/adaptive-lighting/files/11780587/AdaptiveLighting_Configuration.txt)

**Comment by mitchkeller** [comment]
I thought I was going crazy when I could not get this to do what I expected while playing around with sunrise/sunset times and wanting to clear out the manually set values via service call. I am getting the same result here.

**Comment by basnijholt** [comment]
This was a bug, I fixed it in https://github.com/basnijholt/adaptive-lighting/pull/712.

Thanks for reporting.

**Reply by basnijholt** [reply]
(Beta) release with fix is out https://github.com/basnijholt/adaptive-lighting/releases/tag/1.19.0b3
